build.sbt

```diff
@@ -211,6 +214,43 @@ lazy val testStdLayout = Project("daffodil-test-stdLayout", file("test-stdLayout
   .dependsOn(tdmlProc % "test")
   .settings(commonSettings, nopublish)
 
+lazy val xjcSettings =
+  Seq(
+    libraryDependencies ++= Dependencies.xjc,
+    xjcCommandLine ++= Seq(
+      "-nv", 
+      "-p", "org.apache.daffodil.tdml",
+      "-no-header"),
+    xjcLibs ++= Dependencies.xjc,
+    xjcJvmOpts ++= 
+      /* Workaround: certain reflection (used by JAXB) isn't allowed by default in JDK 17:
+      * https://docs.oracle.com/en/java/javase/17/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-7BB28E4D-99B3-4078-BDC4-FC24180CE82B
+      *
+      * While we can handle this JVM quirk at build time, at runtime we won't know
+      * a user's JVM version. We'll need to provide documentation about this.
+      */
+      (if (scala.util.Properties.isJavaAtLeast("17"))
+        Seq(
+          "--add-opens",
+          "java.base/java.lang=ALL-UNNAMED",
+        )
+      else Seq()),
+    Compile / xjc / sources := Seq(
+      file("daffodil-lib/src/main/resources/org/apache/daffodil/xsd/tdml-core.xsd"),
+    ),
+    // Auto-generated Java files have invalid Scaladoc :(
+    Compile / doc / sources ~= (_ filterNot (_.getName endsWith ".java")),
+    // Tests use JAXB, which uses reflection, which has restrictions above JVM 17.
+    Test / fork := true,
```

Comment:
I'm generally not a fan of forking tests, we've seen forking cause things to go much slower. But I guess in this specific case it's only for tdml-lib, which is small enough that it's not really noticeable. So should be fine. Just a heads up to make sure we don't do this globally.

But another win for scalaxb, no need for these Java 17 reflection options (which users running Daffodil on Java 17 would also need to add?), no need to fork tests, no Java conditional dependencies, and no need to mess with doc or xjc sources. Everything just works the way it's expected to.

If xjc is the right technology to use I guess this is fine, but there sure are a lot of hacks being added to get it working that I think just go away if we switch to scalaxb. And with the tdml-core change, the changes are very straightforward. I was able to get it working with just a few settings:

```scala
lazy val scalaxbSettings = Seq(
  Compile / scalaxb / scalaxbPackageName := "org.apache.daffodil.tdml.scalaxb",
  Compile / scalaxb / scalaxbGenerateMutable := true,
  Compile / scalaxb / sources := Seq(
    (lib / Compile / resourceDirectory).value / "org/apache/daffodil/xsd/tdml.xsd",
   ),
  // fixes issue with unused imports
  Compile / scalaxb / scalaxbHttpClientStyle := HttpClientStyle.Sync,
)
```

Granted, I didn't create any tests to see how marshalling/unmarshalling worked, but I imagine it's straightforward.